Code/Framework/AzCore/AzCore/Math/MathUtils.h

```diff
@@ -363,6 +363,21 @@ namespace AZ
         return a > b ? a : b;
     }
 
+    //! Return the gcd of 2 values.
+    //! \note we don't use names like clamp, min and max because many implementations define it as a macro.
+    template<typename T>
+    constexpr T GetGCD(T a, T b)
+    {
+        int c = a % b;
```

Comment:
nit: Since GCD should only operate on integers, you can put a `static_assert` here just in case.
```
static_assert(std::is_integral<T>::value)
```